session/txninfo/txn_info.go

```diff
@@ -65,24 +77,52 @@ type TxnInfo struct {
 	CurrentDB string
 }
 
-// ToDatum Converts the `TxnInfo` to `Datum` to show in the `TIDB_TRX` table
+// Clone clones the TxnInfo. It's safe to call concurrently with the transaction.
```

Comment:
Actually it's not a deep copy and not safe for concurrent modify, maybe clone will confuse the user?

Reply:
Do you mean the `AllSQLDigest` is not cloned but reusing? Actually there's only one thread, which is the transaction's thread, may append to the slice, while there may be other thread reading the same slice, but through another reference. Is this unsafe?
Actually I don't know if there are any better way to do that..

Reply:
It's safe, I think the name is a bit confusing as we don't clone it actually, maybe rename it to `getSnapInfo` or `ref`?

Reply:
Done